openedx/features/enterprise_support/enrollments/utils.py

```diff
@@ -38,32 +45,62 @@ def lms_enroll_user_in_course(
     Arguments:
      - username (str): User name
      - course_id (obj) : Course key obtained using CourseKey.from_string(course_id_input)
-     - mode (CourseMode): course mode
-     - enterprise_uuid (str): id to identify the enterprise to enroll under
+     - desired_mode (CourseMode): desired course mode
+     - enterprise_uuid (str): Optional. id to identify the enterprise to enroll under
      - is_active (bool): Optional. A Boolean value that indicates whether the
```

Comment:
I get that updating based on the value of `is_active` was not the original intention of this change, but now that the function has been renamed to suggest that it can update any updatable field on an enrollment, and `is_active` is still a function argument, why not allow updates to it?  All other layers deep in the enrollment API seem to support updates to both `mode` and `is_active`: https://github.com/edx/edx-platform/blob/3e35e3af8d895c14f05a55a134dbf8a068cd584a/openedx/core/djangoapps/enrollments/data.py#L192

Reply:
you could rename EnrollmentNotUpgradableError -> EnrollmentNotUpdatableError or something else more generic to keep it DRY